docs/lsm/attributes_metadata/attributes_metadata.rst

```diff
@@ -85,3 +85,90 @@ Supported values
 
 
 Attributes modifiers can also be specified on :ref:`relational attributes<attribute_modifiers_on_a_relationship>`.
+
+
+------------
+
+Annotations
+~~~~~~~~~~~
+
+Definition
+##########
+
+Annotations are key-value pairs that can be associated with an entity (service entity or embedded entity) or
+an attribute (simple attribute or relational attribute). These annotations are intended for the web-console to
+improve visualizations and the user-experience.
+
+Annotations on entities
+#######################
+
+Annotations can be attached to an entity using the ``__annotations`` attribute. This attribute has the type ``dict``
+and requires a default value that defines the annotations. Each key-value pair in the dictionary contains respectively
+the name and the value of the annotation. The value of an annotation must always be a string.
+
+Example
+#######
+
+The example below illustrates how the annotation ``annotation=value`` can be set on on a service entity.
+Annotations can be set on embedded entities in the same way.
+
+.. code-block:: inmanta
+
+    entity Interface extends lsm::ServiceEntity:
+        string interface_name
+        dict __annotations = {"annotation": "value"}
+    end
+
+
+Annotations on simple attributes
+################################
+
+Annotations can be attached to simple (non-relational) attributes by defining an attribute of type dict, with a name
+``<attribute>__annotations``, where ``<attribute>`` is the name of the attribute the annotations belong to. This
+attribute needs a default value containing the attributes. The values of the elements in the dictionary must be
+strings.
+
+Example
+#######
+
+The example below shows how the annotation ``annotation=value`` is set on the attribute ``interface_name``.
+Annotations can be set on simple attributes of embedded entities in the same way.
+
+.. code-block:: inmanta
+
+    entity Interface extends lsm::ServiceEntity:
+        string interface_name
+        dict interface_name__annotations = {"annotation": "value"}
+    end
+
+Annotations on relational attributes
+####################################
+
+Annotations can be attach to a relational attribute by replacing the ``--`` part of the relationship definition with
+an instance of the ``lsm::RelationAnnotations`` entity. This entity has a dict attribute ``annotations`` that
+represent the annotation that should be set on the relational attribute. The values of this dictionary must
+be strings. The annotations are always attached to the attribute that is exposed via the lsm API, i.e. the attribute
```

Comment:
I don't get what you mean by this sentence. Are you trying to say that you can't / it doesn't make sense to annotate non-API attributes/relations?

Reply:
me neither

Reply:
I mean: It's a bidirectional relationship, but only the attribute from the parent to the child is part of the API to create/modify/delete the service instance.

Reply:
I'm not entirely sure but I feel like it may sow more confusion than it fixes as it stands now.

Reply:
ok. I will remove it.